src/Control/Effect/Reader.hs

```diff
@@ -39,30 +41,39 @@ instance Effect (Reader r) where
 
 -- | Retrieve the environment value.
 --
---   prop> run (runReader a ask) === a
+-- @
+-- runReader a ('ask' >>= k) = runReader a (k a)
+-- @
 --
 -- @since 0.1.0.0
 ask :: Has (Reader r) sig m => m r
 ask = send (Ask pure)
 
 -- | Project a function out of the current environment value.
 --
---   prop> snd (run (runReader a (asks (applyFun f)))) === applyFun f a
+-- @
+-- 'asks' f = 'fmap' f 'ask'
+-- @
 --
 -- @since 0.1.0.0
 asks :: Has (Reader r) sig m => (r -> a) -> m a
 asks f = send (Ask (pure . f))
 
 -- | Run a computation with an environment value locally modified by the passed function.
 --
---   prop> run (runReader a (local (applyFun f) ask)) === applyFun f a
---   prop> run (runReader a ((,,) <$> ask <*> local (applyFun f) ask <*> ask)) === (a, applyFun f a, a)
+-- @
+-- runReader a ('local' f m) = runReader (f a) m
+-- @
 --
 -- @since 0.1.0.0
 local :: Has (Reader r) sig m => (r -> r) -> m a -> m a
 local f m = send (Local f m pure)
 
 
--- $setup
--- >>> import Test.QuickCheck
--- >>> import Control.Carrier.Reader
+-- Properties
+
+-- | 'ask' returns the environment variable.
+--
+-- @since 1.0.0.0
+ask_environment :: Has (Reader r) sig m => (b -> b -> prop) -> (r -> m a -> b) -> r -> (r -> m a) -> prop
```

Comment:
Just curious, why you are snake case over camel case here? There must be a convention I miss.

Reply:
It’s the convention used for e.g. QuickCheck properties to make them easily discoverable using Template Haskell.